### PR TITLE
Feat/spec renderer resolved

### PR DESCRIPTION
**Pull request by jackkav**
Motivation: use internel spec renderer, over swagger-ui-dist
Aim: drop-in replacement
Todo:
- [x] add and use @kong/spec-renderer
- [ ] remove swagger-ui-dist
- [ ] point at spec-renderer stable build
- [ ] confirm behaviour and layout meets aim
 - [ ] scroll overflow
 - [x] color themeing decision, either dark/light or take something similar to our css variable palette

Future work
- use bundler type module resolution in insomnia

### CODE REVIEW

**Comment by adamdehaven** [comment]
> color themeing decision, either dark/light or take something similar to our css variable palette

@jackkav for now, we'd recommend utilizing the default theme (light) in the short term. There is theming available currently via CSS Custom Properties; however, we're going to bake-in easier theming and possibly a default dark theme that you can then tweak for Insomnia specifically. We're working with Design on getting some "official" themes, so recommended to hold on working on customizing on your own in the short-term

**Comment by jackkav** [comment]
I was trying to fix the styling and I encountered a couple of dx issues, 
1. the vite dev server started opening a chrome tab on save
2. a new error is appearing in the console `Warning: You are calling ReactDOMClient.createRoot() on a container that has already been passed to createRoot() before. Instead, call root.render() on the existing root instead if you want to update it.`

**Comment by adamdehaven** [comment]
> I was trying to fix the styling and I encountered a couple of dx issues,
> 
> 1. the vite dev server started opening a chrome tab on save

I'm not sure on this one; nothing in the web component should be triggering this?

> 2. a new error is appearing in the console `Warning: You are calling ReactDOMClient.createRoot() on a container that has already been passed to createRoot() before. Instead, call root.render() on the existing root instead if you want to update it.`

This typically happens when a `DOMContentLoaded` event listener is triggered twice. Is there any chance [this line](https://github.com/Kong/insomnia/blob/2fdd6b19695b8e6c0fa1f4e4e98ee5de2654abed/packages/insomnia/src/ui/index.tsx#L1162) is getting called more than once?

Additionally, do you need to unmount the web component at some point, e.g. `root.unmount()`?

File: packages/insomnia/package.json [hunk 1]
```diff
@@ -42,6 +42,7 @@
     "@getinsomnia/node-libcurl": "^2.4.31-0",
     "@grpc/grpc-js": "^1.10.10",
     "@grpc/proto-loader": "^0.7.13",
+    "@kong/spec-renderer-dev": "^1.76.4-pr.339.fdf5330.0",
```
Comment (adamdehaven) [inline packages/insomnia/package.json:45]: We merged in the changes from this PR preview into `main` for now (still need to resolve exports, etc.) but you can install the latest

`npm i @kong/spec-renderer-dev@latest`

**Comment by jackkav** [comment]
Both issues could be caused by the place we register I can try a few other places that are more likely to only eval once

**Comment by jackkav** [comment]
above idea didn't work, looks like the spec renderer register function is interfering with vite HMR in insomnia.
moving discussion to https://github.com/Kong/insomnia/pull/8036

**Pull request closed without merging**